source/agora/utils/TxBuilder.d

```diff
@@ -155,10 +155,12 @@ public struct TxBuilder
     public ref typeof(this) attach (const Transaction tx)
         @safe nothrow return
     {
+        // Don't include small change outputs less than required fees
         this.inputs ~= iota(tx.outputs.length)
             .map!(index => OutputRef(tx.outputs[index], Input(tx, index).utxo))
+            .filter!(o => o.output.value > Amount(5000))
```

Comment:
Why ? That seems rather arbitrary. I think the caller should be at liberty to do this. It's obviously wasteful to do so, but the job of `TxBuilder` is to build a transaction, not to define best practices for doing so.
So unless this would systematically lead to invalid transactions being created, I don't see a reason to do it.
Moreover, doing it *silently* would lead to hard to debug issues, where a user wrongly uses `TxBuilder` and gets no feedback (imagine the user having only <= 5k amounts outputs, trying to use TxBuilder, and seeing it empty).

Reply:
I was forgetting that you can attach multiple times to end up with enough in leftover to cover the fees. So this will be reverted and the calling code will have to handle the cases where the output is small change and would create invalid transactions.